packages/chains/src/chains/chainMetadata/index.ts

```diff
@@ -160,3 +236,31 @@ export const chains: {
     nativeToken: eth,
   },
 };
+
+const MS_PER_DAY = 1000 * 60 * 60 * 24;
+
+export const getBlockTimeByChainId = (chainId: ChainId, targetTimestamp = new Date().getTime()) => {
+  const targetChain = chains?.[chainId ?? ''];
+  if (!targetChain || !Array.isArray(targetChain.blockTimes)) return null;
+
+  // default to latest
+  let blockTime = targetChain.blockTimes?.slice(-1)?.[0];
```

Comment:
```suggestion
  let blockTime = targetChain.blockTimes[0];
```

Reply:
You don't need to default to the latest block time since the block right after would find the right block time by going through the array.